### ISSUE TITLE
Persistence flushing with $forget=true won't work as expected in some cases.

### ISSUE THREAD

**Issue by kevindierkx** [cartalyst/sentinel#278]
Looking at the `IlluminatePersistenceRepository` we have the following flush implementation:
```PHP
public function flush(PersistableInterface $persistable, $forget = true)
{
    if ($forget) {
        $this->forget($persistable);
    }
    foreach ($persistable->{$persistable->getPersistableRelationship()}()->get() as $persistence) {
        if ($persistence->code !== $this->check()) {
            $persistence->delete();
        }
    }
}
```

This accepts `$forget` as the second parameter and defaults to true, to my understanding this is meant to flush all persistences for a `PersistableInterface` and forget the session for this entity.

The problem is the `flush` method, it doesn't accept a `PersistableInterface` and will always kill the current session. Which CAN belong to the same `PersistableInterface` but it also CAN NOT.

```PHP
public function forget()
{
    $code = $this->check();

    if ($code === null) {
        return;
    }

    $this->session->forget();
    $this->cookie->forget();

    return $this->remove($code);
}
```

Long story short, I don't really know what the intend of this specific piece of code is, but I'm certain it doesn't behave as expected at the moment.

**Comment by suwardany** [comment]
Hello @kevindierkx 

Thanks for reporting.

That argument shouldn't be passed to the `forget` method, i've pushed a fix for that (doesn't really affect behavior)

Incase you want to flush persistences for a different entity (other that your current logged in user) .. you simply have to pass false as the second argument, since your current session has nothing to do with that of that user, you do not need that flushed.

Let me know if you need any further information.
